File: simple-stack-example-fragments/src/main/java/com/zhuinden/simplestackdemoexamplefragments/data/entity/DbTask.java
package com.zhuinden.simplestackdemoexamplefragments.data.entity;

import io.realm.RealmObject;
import io.realm.annotations.Index;
import io.realm.annotations.PrimaryKey;

/**
 * Created by Owner on 2017. 01. 26..
 */

public class DbTask
        extends RealmObject {
    @PrimaryKey
    private String id;

    private String title;

    private String description;

    @Index
    private boolean completed;

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public boolean getCompleted() {
        return completed;
    }

    public void setCompleted(boolean completed) {
        this.completed = completed;
    }
}

File: simple-stack-example-mvp-fragments/src/main/java/com/zhuinden/simplestackdemoexamplefragments/data/entity/DbTask.kt
package com.zhuinden.simplestackdemoexamplefragments.data.entity

import io.realm.RealmObject
import io.realm.annotations.Index
import io.realm.annotations.PrimaryKey

/**
 * Created by Owner on 2017. 01. 26..
 */

open class DbTask : RealmObject() {
    @field:PrimaryKey
    var id: String? = null

    var title: String? = null

    var description: String? = null

    @field:Index
    var completed: Boolean = false
}
